commands/getStats.js

```diff
@@ -0,0 +1,233 @@
+  
+    return [
+      null,
+      `
+  \`\`\`
+  ${majorTable.toString()}
+  \`\`\`
+      `
+    ];
+  },
+
+
+
+  // get count of each graduation year
+  // linked to STATS command
+  "getYearStats" : async function () {
+    // open db
+    let db = await sqlite.open({
+      filename: config.db_path,
+      driver: sqlite3.Database,
+    });
+  
+    // get count of each year
+    let rows = null;
+    try {
+      rows = await db.all(
+        `
+  SELECT
+    grad_year, count(*) 'count'
+  FROM users
+  GROUP BY grad_year
+  ORDER BY count DESC`
+      );
+    } catch (e) {
+      console.error(e.toString());
+      await db.close();
+      return [{ message: e.toString() }, null];
+    }
+    if (!rows) {
+      return [
+        null,
+        `
+  No stats available on graduation years.`,
+      ];
+    }
+  
+    let yearTable = new AsciiTable('Graduation Year by Count');
+    yearTable.setHeading('Year','Count');
+    rows.forEach(row => yearTable.addRow(row.grad_year, row.count));
+  
+    return [
+      null,
+      `
+  \`\`\`
+  ${yearTable.toString()}
+  \`\`\`
+      `
+    ];
+  },
+
+  // get number of transfers
+  // linked to STATS command
+  "getNumTransferStats" : async function () {
+    // open db
+    let db = await sqlite.open({
+      filename: config.db_path,
+      driver: sqlite3.Database,
+    });
+  
+    // get number of transfer students
+    let row = null;
+    try {
+      row = await db.get(
+        `
+  SELECT
+    count(*) 'count'
+  FROM users
+  WHERE transfer_flag = 1`
+      );
+    } catch (e) {
+      console.error(e.toString());
+      await db.close();
+      return [{ message: e.toString() }, null];
+    }
+    if (!row) {
+      return [
+        null,
+        `
+  No stats available on transfer students.`,
+      ];
+    }
+  
+    return [
+      null,
+      `
+  \`\`\`
+  Number of Transfer Students: ${row.count}
+  \`\`\`
+      `
+    ];
+  },
+
+
+  // get count of each affiliation
+  // linked to STATS command
+  "getAffiliationStats" : async function () {
+    // open db
+    let db = await sqlite.open({
+      filename: config.db_path,
+      driver: sqlite3.Database,
+    });
+  
+    // get count of each affiliation
+    let rows = null;
+    try {
+      rows = await db.all(
+        `
+  SELECT
+    affiliation, count(*) 'count'
+  FROM users
+  GROUP BY affiliation
+  ORDER BY count DESC`
+      );
+    } catch (e) {
+      console.error(e.toString());
+      await db.close();
+      return [{ message: e.toString() }, null];
+    }
+    if (!rows) {
+      return [
+        null,
+        `
+  No stats available on affiliation.`,
+      ];
+    }
+  
+    let affilTable = new AsciiTable('Affiliation by Count');
+    affilTable.setHeading('Affiliation','Count');
+    rows.forEach(row => affilTable.addRow(row.affiliation, row.count));
+  
+    return [
+      null,
+      `
+  \`\`\`
+  ${affilTable.toString()}
+  \`\`\`
+      `
+    ];
+  }
+}
```

Comment:
You should configure your editor to add a newline at the end of file. Honestly, this is just a nitpick but it's also standard. If you're wondering why this matters, it's only usually important when you need to concatenate files or similar reasons. There's some definitional play too lol